p2p/trust/trustmetric.go

```diff
@@ -124,6 +124,37 @@ type peerHistoryJSON struct {
 	History      []float64 `json:"history"`
 }
 
+// Loads the history data for a single peer and takes care of trust metric locking
+func reinstantiateMetric(tm *TrustMetric, ph peerHistoryJSON) {
```

Comment:
This might be a bit cleaner as an `Init` method on `TrustMetric`. 
Also, since it's only called once and only right after creation, it shouldn't actually need the lock. Comments should reflect this.